firefox-ios/firefox-ios-tests/Tests/ClientTests/Coordinators/Mocks/MockStoreForMiddleware.swift

```diff
@@ -53,7 +53,12 @@ class MockStoreForMiddleware<State: StateType>: DefaultDispatchStore {
         // TODO: if you need it
     }
 
-    func dispatch(_ action: Redux.Action) {}
+    func dispatch(_ action: Redux.Action) {
```

Comment:
Curious on the decision behind using a `lock`?  Was using `@MainActor` considered, and it didn't work? In my opinion, it might be better to reflect the actor context as we do in the real cod if we could.

cc: @yoanarios

Reply:
I think this is just adding the mock code as we add inside the `dispatchLegacy` function below. I did the same in [this PR](https://github.com/mozilla-mobile/firefox-ios/pull/28178/files#diff-652166053aae86b3a1c8a8cd64e9ec3e0fab89f404ae0308393286c3d0a9f11c) (already merged). We can probably rethink the lock usage if it's `@MainActor` as you said?

Reply:
Ah previously it was a simple fix to ensure thread safety, and now I wondering with modern swift concurrency if there is a better way. So maybe we can revisit how we handle thread safety in the mocks? Since it's merged, we can revisit, but maybe add a `TODO` to investigate and think about it more 🤔 ?